### DISCUSSION TITLE
Issues migrating from `load_assets_from_dbt_project` and `dbt_cli_resource` to `@dbt_assets` and `DbtCliResource`

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17633] [Q&A] [answered]
Hi folks,
I'm migrating from `load_assets_from_dbt_project(` and `dbt_cli_resource` to `@dbt_assets` and `DbtCliResource` and I'm running into some issues.
1. My assest are no longer automatically grouped and I can't find how to group them depending on their folder
2. I sometimes need to specify a dbt variable or  do a full-refresh of some assets when I run a job. It was really easy to do it from the interface with the `dbt_cli_resource`  but I no longer find these options with `DbtCliResource`.
3. With the `stream()` method, the logs are less precise. I've seen `stream_raw_events()` in the doc but I can't make it work.  This is a problem for me because I've just had a successful materialization on the interface, but in the dbt logs in my terminal, I have the following message : `The selection criterion 'fqn:dbt_project.group.sub_group.table' does not match any nodes.` The model actually exists,but there may be a problem with the selection of models in the subgroups.

Could someone help me with this?
Thank you

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C04CW71AGBW/p1698927516264959?thread_ts=1698927516.264959&cid=C04CW71AGBW).*

**Comment by rexledesma** [comment, marked as answer]
> 1. My assets are no longer automatically grouped and I can't find how to group them depending on their folder

By default, Dagster group names no longer use the dbt folder structure to determine the group name. See https://docs.dagster.io/integrations/dbt/reference#customizing-group-names for more details.

To get the old behavior, you can define a custom [`DagsterDbtTranslator`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DagsterDbtTranslator) and use the [`group_from_dbt_resource_props_fallback_to_directory`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.group_from_dbt_resource_props_fallback_to_directory) function. 

```python
from typing import Any, Mapping, Optional

from dagster import (
    AssetExecutionContext,
)
from dagster_dbt import (
    DagsterDbtTranslator,
    DbtCliResource,
    dbt_assets,
    group_from_dbt_resource_props_fallback_to_directory,
)

from .constants import dbt_manifest_path


class CustomDagsterDbtTranslator(DagsterDbtTranslator):
    @classmethod
    def get_group_name(cls, dbt_resource_props: Mapping[str, Any]) -> Optional[str]:
        return group_from_dbt_resource_props_fallback_to_directory(dbt_resource_props)


@dbt_assets(manifest=dbt_manifest_path)
def jaffle_shop_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from dbt.cli(["build"], context=context).stream()

```

> 2. I sometimes need to specify a dbt variable or do a full-refresh of some assets when I run a job. It was really easy to do it from the interface with the dbt_cli_resource but I no longer find these options with DbtCliResource.

This is covered in the API examples of [`@dbt_assets`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.dbt_assets). You should add custom [`Config`](https://docs.dagster.io/concepts/configuration/config-schema) to your assets if you plan on adding dbt vars or a `--full-refresh` from the Dagster UI.

For example, adding config to do a `--full-refresh`:

```python
from pathlib import Path

from dagster import AssetExecutionContext, Config
from dagster_dbt import DagsterDbtTranslator, DbtCliResource, dbt_assets


class MyDbtConfig(Config):
    full_refresh: bool


@dbt_assets(manifest=Path("target", "manifest.json"))
def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource, config: MyDbtConfig):
    dbt_build_args = ["build"]
    if config.full_refresh:
        dbt_build_args += ["--full-refresh"]

    yield from dbt.cli(dbt_build_args, context=context).stream()
```

> 3. With the stream() method, the logs are less precise. I've seen stream_raw_events() in the doc but I can't make it work. This is a problem for me because I've just had a successful materialization on the interface, but in the dbt logs in my terminal, I have the following message : The selection criterion 'fqn:dbt_project.group.sub_group.table' does not match any nodes. The model actually exists,but there may be a problem with the selection of models in the subgroups.

We are emitting the same logs as the ones from dbt, so I am not sure how these are less precise. These are viewable in the `stdout` tab in the Dagster run.

**Reply by AugustinWBD** [reply]
Thank you very much for your complete answer.

For the 3rd point, indeed, I didn't see the full logs in stdout.
But do you have an idea why dbt can't find my asset ?
